common/bucket_utils.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for interacting with the buckets."""
+
+from common import logs
+from common import experiment_utils
+
+logger = logs.Logger('bucket_utils')
+
+
+def __using_gsutil():
+    """ Test whether using gsutil format paths. """
+    try:
+        experiment_path_format = experiment_utils.get_cloud_experiment_path()
+    except KeyError:
+        experiment_path_format = 'gs://default'
+
+    return experiment_path_format.startswith('gs://')
+
+
+if __using_gsutil():
+    from common import gsutil as base_utils
+else:
+    # TODO(zhichengcai): local_utils
+    from common import gsutil as base_utils
+
+
+def command(arguments, *args, parallel=False, **kwargs):
```

Comment:
I don't think this should be a public function in either of the submodules.

Reply:
Currently, there are test function calling gsutil.command, which is the previous gsutil.gsutil_command.

Reply:
I prefer to have this interface. Though current code doesn't have callings to `gsutil.command()`, it makes sense to leave such interface for users to call.

Reply:
I don't agree that it makes sense for users to call this and I don't find the testing reason convincing either. Please remove.